programs.cc

```diff
@@ -154,13 +155,17 @@ bool RunInstance::start( Config::Program const & prg, QString const & word,
     args.pop_front();
 
     bool writeToStdInput = true;
+    auto const & search_string = GlobalBroadcaster::instance()->translateLineText;
 
-    for( int x = 0; x < args.size(); ++x )
-      if( args[ x ].indexOf( "%GDWORD%" ) >= 0 )
-      {
+    for( auto & arg : args ) {
+      if( arg.indexOf( "%GDWORD%" ) >= 0 ) {
         writeToStdInput = false;
-        args[ x ].replace( "%GDWORD%", word );
+        arg.replace( "%GDWORD%", word );
+      }
+      if( arg.indexOf( "%GDSEARCH%" ) >= 0 ) {
+        arg.replace( "%GDSEARCH%", search_string );
       }
+    }
 
     process.start( programName, args );
     if( writeToStdInput )
```

Comment:
if %GDSEARCH% is provided ,   the writeToStdInput will be remain true.
and will make the %GDWORD% send to the std again
`if( writeToStdInput )`

Reply:
> if %GDSEARCH% is provided , the writeToStdInput will be remain true.

I thought this should be the expected behavior since %GDSEARCH% is never sent to the stdin. The program then can decide whether to read %GDWORD% from stdin or ignore it.

But I can change it so that `writeToStdInput = false;` is applied in the second case as well.